app/src/main/java/org/oppia/android/app/translation/AppLanguageResourceHandler.kt

```diff
@@ -151,18 +152,38 @@ class AppLanguageResourceHandler @Inject constructor(
    */
   fun computeLocalizedDisplayName(audioLanguage: AudioLanguage): String {
     return when (audioLanguage) {
-      AudioLanguage.HINDI_AUDIO_LANGUAGE -> getLocalizedDisplayName("hi")
-      AudioLanguage.FRENCH_AUDIO_LANGUAGE -> getLocalizedDisplayName("fr")
-      AudioLanguage.CHINESE_AUDIO_LANGUAGE -> getLocalizedDisplayName("zh")
-      AudioLanguage.BRAZILIAN_PORTUGUESE_LANGUAGE -> getLocalizedDisplayName("pt", "BR")
+      AudioLanguage.HINDI_AUDIO_LANGUAGE ->
+        resources.getString(R.string.hindi_localized_language_name)
+      AudioLanguage.FRENCH_AUDIO_LANGUAGE ->
+        resources.getString(R.string.french_localized_language_name)
+      AudioLanguage.CHINESE_AUDIO_LANGUAGE ->
+        resources.getString(R.string.chinese_localized_language_name)
+      AudioLanguage.BRAZILIAN_PORTUGUESE_LANGUAGE ->
+        resources.getString(R.string.brazilian_portuguese_localized_language_name)
       AudioLanguage.NO_AUDIO, AudioLanguage.AUDIO_LANGUAGE_UNSPECIFIED, AudioLanguage.UNRECOGNIZED,
-      AudioLanguage.ENGLISH_AUDIO_LANGUAGE -> getLocalizedDisplayName("en")
+      AudioLanguage.ENGLISH_AUDIO_LANGUAGE
+      -> resources.getString(R.string.english_localized_language_name)
     }
   }
 
-  private fun getLocalizedDisplayName(languageCode: String, regionCode: String = ""): String {
-    // TODO(#3791): Remove this dependency.
-    val locale = Locale(languageCode, regionCode)
-    return locale.getDisplayLanguage(locale).capitalize(locale)
+  /**
+   * Returns a human-readable, localized representation of the specified [OppiaLanguage].
+   * Note that the returned string is not expected to be localized to the user's current locale.
+   * Instead, it will be localized for that specific language (i.e. each language will be
+   * represented within that language to make it easier to identify when choosing a language).
+   */
+  fun computeLocalizedDisplayName(oppiaLanguage: OppiaLanguage): String {
+    return when (oppiaLanguage) {
+      OppiaLanguage.HINDI -> resources.getString(R.string.hindi_localized_language_name)
+      OppiaLanguage.PORTUGUESE ->
+        resources.getString(R.string.portuguese_localized_language_name)
+      OppiaLanguage.SWAHILI -> resources.getString(R.string.swahili_localized_language_name)
+      OppiaLanguage.BRAZILIAN_PORTUGUESE
+      -> resources.getString(R.string.brazilian_portuguese_localized_language_name)
```

Comment:
```suggestion
      OppiaLanguage.BRAZILIAN_PORTUGUESE ->
        resources.getString(R.string.brazilian_portuguese_localized_language_name)
```

Line-breaking is done at the higher syntactical level.

Reply:
Updated, thanks.